code/citadel/hypospraymkii.dm

```diff
@@ -0,0 +1,238 @@
+			to_chat(user, "<span class='notice'>You inject [amount_per_transfer_from_this] units of the solution. The hypospray's cartridge now contains [reagents.total_volume] units.</span>")
+
+	if(HYPO_SPRAY)
+		//Always log attemped injections for admins
+		var/list/rinject = list()
+		for(var/datum/reagent/R in reagents.reagent_list)
+			rinject += R.name
+		var/contained = english_list(rinject)
+		add_logs(user, L, "attemped to spray", src, addition="which had [contained]")
+
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='notice'>[src]'s cartridge is empty.</span>")
+			return
+
+		if(target.reagents.total_volume >= target.reagents.maximum_volume)
+			to_chat(user, "<span class='notice'>[target] is full.</span>")
+			return
+
+		if(L) //living mob
+			if(!L.can_inject(user, TRUE))
+				return
+			if(L != user)
+				L.visible_message("<span class='danger'>[user] is trying to spray [L] with [src]!</span>", \
+									"<span class='userdanger'>[user] is trying to spray [L] with the [src]!</span>")
+				if(!do_mob(user, L, extra_checks=CALLBACK(L, /mob/living/proc/can_inject,user,1)))
+					return
+				if(!reagents.total_volume)
+					return
+				if(L.reagents.total_volume >= L.reagents.maximum_volume)
+					return
+				if(src.possible_transfer_amounts >= 15)
+					playsound(loc,'sound/items/hypospray_long.ogg',50, 1, -1)
+				else
+					playsound(loc, "inject_sounds", 50, 1, -1)
+				L.visible_message("<span class='danger'>[user] injects [L] with the [src]!", \
+								"<span class='userdanger'>[user] injects [L] with the [src]!</span>")
+
+			if(L != user)
+				add_logs(user, L, "sprayed", src, addition="which had [contained]")
+			else
+				if(src.possible_transfer_amounts >= 15)
+					playsound(loc,'sound/items/hypospray_long.ogg',50, 1, -1)
+				else
+					playsound(loc, "inject_sounds", 50, 1, -1)
+				log_attack("<font color='red'>[user.name] ([user.ckey]) sprayed [L.name] ([L.ckey]) with [src.name], which had [contained] (INTENT: [uppertext(user.a_intent)])</font>")
+				L.log_message("<font color='orange'>Sprayed themselves ([contained]) with [src.name].</font>", INDIVIDUAL_ATTACK_LOG)
+		var/fraction = min(amount_per_transfer_from_this/reagents.total_volume, 1)
+		reagents.reaction(L, TOUCH, fraction)
+		reagents.trans_to(target, amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You spray [amount_per_transfer_from_this] units of the solution. The hypospray's cartridge now contains [reagents.total_volume] units.</span>")
+
+/obj/item/reagent_containers/hypospray/mkii/AltClick(var/obj/item/reagent_containers/glass/bottle/vial/W, mob/living/user)
+	testing("Altclick registered, check mob stat")
+	if(user.incapacitated() || !istype(user))
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+	testing("Check for hypo")
+	if(!contents)
+		to_chat(usr, "This Hypo needs to be loaded first!")
+		return
+	testing("Has hypo")
+	var/obj/item/reagent_containers/glass/bottle/vial/W
```

Comment:
There's one issue

You're defining the var here, but you aren't setting it to anything

Reply:
it doesn't even progress this chain anyway. it stops at being registered.